pkg/cmd/addon.go

```diff
@@ -120,129 +150,33 @@ func NewAddonListCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Comma
 	return cmd
 }
 
-func retrievePlugins(ctx context.Context, c client.Client, ioStreams cmdutil.IOStreams, namespace string) error {
-	var pluginList []Plugin
-	var config AddonCenterConfig
-	var data []byte
-	var err error
-	var homeDir string
-	if homeDir, err = os.UserHomeDir(); err != nil {
-		ioStreams.Errorf("Failed to retrieve addon center configuration, please run `vela addon:config` first")
-	}
-	if data, err = ioutil.ReadFile(filepath.Join(homeDir, addonCenterConfigFile)); err != nil {
-		ioStreams.Errorf("Failed to retrieve addon center configuration, please run `vela addon:config` first")
-	}
-	if err := json.Unmarshal(data, &config); err != nil {
-		ioStreams.Errorf("Failed to retrieve addon center configuration, please run `vela addon:config` first")
+func ListRepoAddons(repoDir string, ioStreams cmdutil.IOStreams) error {
+	templates, err := plugins.LoadTempFromLocal(repoDir)
+	if err != nil {
+		return err
 	}
+	table := uitable.New()
+	table.AddRow("NAME", "TYPE", "DEFINITION", "STATUS", "APPLIES-TO")
 
-	if config.IsLocal {
-		//TODO(zzxwill) merge `vela traits` and `vela workloads`
-		return nil
-	} else {
-		resp, err := http.Get(config.Name)
-		if err != nil {
-			return err
-		}
-		defer resp.Body.Close()
-		result, _ := ioutil.ReadAll(resp.Body)
-		var manifests []PluginFile
-		var traitManifestPrefix, workloadManifestPrefix = "TraitDefinition", "WorkloadDefinition"
-		err = json.Unmarshal(result, &manifests)
-		if err != nil {
-			return err
-		}
-		var manifestResp *http.Response
-		for _, d := range manifests {
-			var template types.Template
-			var workloadDefinition corev1alpha2.WorkloadDefinition
-			var traitDefinition corev1alpha2.TraitDefinition
-			if manifestResp, err = http.Get(d.Url); err != nil {
-				return err
-			}
-			defer manifestResp.Body.Close()
-
-			if strings.Contains(strings.ToLower(d.Name), strings.ToLower(workloadManifestPrefix)) {
-				result, err := ioutil.ReadAll(manifestResp.Body)
-				if err != nil {
-					return err
-				}
-				err = yaml.Unmarshal(result, &workloadDefinition)
-				if err != nil {
-					return err
-				}
-				var definitionName string
-				if workloadDefinition.Spec.Extension != nil {
-					template, err = types.ConvertTemplateJson2Object(workloadDefinition.Spec.Extension)
-					if err != nil {
-						return err
-					}
-					definitionName = template.Name
-				} else {
-					definitionName = workloadDefinition.Name
-				}
-
-				if err != nil {
-					return err
-				}
-				//Check whether the definition is applied
-				var status = "uninstalled"
-				if _, err = cmdutil.GetWorkloadDefinitionByName(ctx, c, namespace, workloadDefinition.Name); err == nil {
-					status = "installed"
-				}
```

Comment:
Is it the right way to check the installation status? If yes, I can fulfill the TODO after the merge:)

Reply:
yes, but we should get definition from `.rudr/definitions/` instead of from cluster